M2/Macaulay2/m2/document.m2

```diff
@@ -550,7 +550,7 @@ getSourceCode :=  val         -> DIV {"class" => "waystouse",
     fixup DIV {SUBSECTION "Code", PRE M2CODE demark_newline unstack stack apply(enlist val, m -> (
 		f := lookup m; if f === null then error("SourceCode: ", toString m, ": not a method");
 		c := code f;   if c === null then error("SourceCode: ", toString m, ": code for method not found");
-		reproduciblePaths toString c))}}
+		reproduciblePaths toString net c))}}
```

Comment:
Might it be easier to change `toString Hypertext` to do `toString net x`, and use `toExternalString Hypertext` to get what `toString Hypertext` currently does?

Reply:
I'm not sure that's a good idea. right now `toString` and `net` have very different outputs. here I do mean, take the usual `net` output, but turn it into a string (among other things, to be fed into `reproduciblePaths`, which itself is a bit of a hack and I will rewrite in a future PR).

Reply:
`toString` and `net` are far closer than `toExternalString`. For instance, what's the point of `toString PRE "hi"` returning `PRE{hi}`? That output can't be used anywhere else, can it?

Reply:
I don't disagree with you, but maybe this is outside the scope of this PR. The standard behavior of `toString` for pretty much any type of `BasicList` is to write the name of the type followed by the contents. I'm not ready to change this at this stage, because it may have unintended consequences.

Reply:
Is there a descendant of `BasicList` where `net` and `toString` print different things? I think in all the types you're referring to `net` also writes the name of the type followed by the contents, so if `net Hypertext` knows how to properly format the object, I definitely think `toString Hypertext` should also do that.

Reply:
again, not disagreeing. I'll make a note of it and try at some point.